src/httpc.c

```diff
@@ -145,9 +145,12 @@ httpc_env_destroy(struct httpc_env *ctx)
 {
 	assert(ctx);
 
-	curl_env_destroy(&ctx->curl_env);
-
-	mempool_destroy(&ctx->req_pool);
+	if (mempool_count(&ctx->req_pool) == 0) {
+		mempool_destroy(&ctx->req_pool);
+		curl_env_destroy(&ctx->curl_env);
+	} else {
+		ctx->curl_env.cleanup = 1;
+	}
```

Comment:
I don't see a path of execution when the `curl_env_destroy(&ctx->curl_env);` is called with `ctx->curl_env.cleanup == 1`. There could be a memory leak here.

Reply:
Patch has been significantly updated.